### PR TITLE
use httpsnoop to ensure http.ResponseWriter additional interfaces are preserved

### PR DESCRIPTION
**Pull request by toshok**
Noticed this problem when going through the websocket upgrade flow.  http.Hijacker is implemented by builtin response writers but not by the recordingResponseWriter.

Instead of doing all the leg work to ensure that interfaces are implemented correctly, convert things to use httpsnoop, which does codegen to generate the possible combinations of interface implementations so that things are preserved.

One question: given that the `recordingResponseWriter` initialization code now includes an additional set of allocations (from `httpsnoop`, but also the closures for the hooks), is it worth it to keep the sync.Pool?  My gut says no, but... ?

### CODE REVIEW

File: go.mod
Comment (MrAlias) [inline go.mod]: These look like they were inadvertent. The `otelmux` module should be the only place these dependencies need to be added.

**Review by MrAlias — APPROVED** [review]

File: go.mod
Comment (toshok) [reply go.mod]: yep, I failed to run `make precommit` 👍

**Comment by toshok** [comment]
hm, seeing FD exhaustion in a service running with this now.  recommend holding off on accepting/landing

**Comment by toshok** [comment]
phew, tracked down FD leak - not anything related to this change, so should be good to go

**Review by lizthegrey — APPROVED** [review]

**Comment by MrAlias** [comment]
@toshok This looks ready to merge given your validation. It appears maintainer edits are off for this so I cannot sync with master. Can you sync so we can merge this?

**Comment by toshok** [comment]
rebased off current master

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**
